ir/src/lib.rs

```diff
@@ -36,12 +36,21 @@ impl AirIR {
     /// Consumes the provided source and generates a matching AirIR.
     pub fn from_source(source: &ast::Source) -> Result<Self, SemanticError> {
         let ast::Source(source) = source;
+
+        // ensure that the source contains trace columns section
+        if !contains_trace_cols(source) {
+            return Err(SemanticError::MissingSection(
+                "Trace Columns section is missing".to_string(),
```

Comment:
nit: I would prefer to use the syntax of the missing section, e.g. "'trace_columns' section is missing" or "Trace Columns section `trace_columns` is missing"